internal/observability/event/options.go

```diff
@@ -70,6 +71,13 @@ func WithFlush() Option {
 	}
 }
 
+// WithInfo allows an optional map as info
+func WithInfo(d map[string]interface{}) Option {
```

Comment:
Looking at the usage of this function and how it changes the existing code, this introduces some very long lines, which I find reduces the readability of the code.

For example compare these two lines:
```go
s.logger.Error("error creating job repo", "error", err)
event.WriteError(ctx, op, err, event.WithInfo(map[string]interface{}{"msg": "error creating job repo"}))
```

In particular I think the `map[string]interface{}` is distracting. Perhaps a type to shorten this:

```go
// in event/options.go
type Info map[string]interface{}

// in scheduler/scheduler.go
event.WriteError(ctx, op, err, event.WithInfo(event.Info{"msg": "error creating job repo"}))
```

Or even shorter:

```go
// in event/options.go
type I map[string]interface{}

// in scheduler/scheduler.go
event.WriteError(ctx, op, err, event.WithInfo(event.I{"msg": "error creating job repo"}))
```

I also notice that `options.withDetails` and `options.WithHeader` are also `map[string]interface{}`. Defining types could help clarify if there are any conceptual differences between them:

```go
// separate types to be explicit and make it clear that Details are different
// from Info
type Info map[string]interface{}
type Detail map[string]interface{}
type Header map[string]interface{}

// or a shorter version
type I map[string]interface{}
type D map[string]interface{}
type H map[string]interface{}

// a single type to show that they are all the same kv data, and it is the With
// functions that provide the distinction.
type KeyValue map[string]interface{}
// or perhaps
type KV map[string]interface{}
```

Also, it looks like there are a lot of places where the `Info` is just a single key of `msg`. It might be useful to have a conience function for this:

```go
// in event/options.go
func WithInfoMsg(msg string) Option {
    return WithInfo(map[string]interface{
      "msg": msg,
    })
}

// in scheduler/scheduler.go
event.WriteError(ctx, op, err, event.WithInfoMsg("error creating job repo"}))
```

That said, this might be a conversation for a separate PR, since it would impact code that isn't changing in this PR.

Reply:
Hum... I'd like to incorporate any changes to these options now.. since they would ripple across all the code base where ever they're used. 

I like the `WithInfoMsg(...)` suggestion and will add that.  ty!

I'm not sure adding `KV` adds any clarity.

Pondering the whole set of types for `I, D, H`.

Reply:
`WithInfoMsg` does seem to be a convenient helper if we foresee that most Info maps will be a single key of `msg`

Reply:
I've adopted your suggestions about `I, D, H` and I've include using `I` in `WriteSysEvent(...)` as well.  ty!

Reply:
A lot of libs that take in this kind of data use a `...args interface{}` kind of pattern that expect a key, value, key, value kind of pattern, and check that it matches and that the keys are strings. `hclog` itself is an example of this. So another possibility is something like `event.WithInfo("mykey", 1, "other key", false)`